# Python Programming Basic 2 실전

## URL 주소를 만들어주세요
- playdata를 입력했을 때, https://playdata.io 를 반환하는 함수를 만들어 주세요.
    - playdata_url('playdata')

In [20]:
# 함수 작성
def playdata_url(playdata) : #wksk
  playdata = 'playdata'
  return 'https://playdata.io'

In [28]:
def playdata_url2(name):
  return 'https://' + name + '.io'

playdata_url('playdata')

'https://playdata.io'

In [10]:
print(playdata_url('playdata')) #wksk

https://playdata.io


### 결과
'https://playdata.io'

## 숫자만을 원해
- 숫자를 다루어야 하는 일이 생겼습니다. 그런데 데이터를 보니 콤마가 포함이 되어있네요... 나는 숫자만 필요한데... 숫자만 반환해주는 함수를 만들어 주세요.
    - remove_comma(data)

In [ ]:
data = '23,524,414'

In [24]:
# 함수 작성
def remove_comma(data):
  return int(data.replace(',',''))

In [26]:
remove_comma('23,524,414')

23524414

In [27]:
# 2-2
def remove_comma2(data):
  data_new = data.split(',')
  num = ''
  for i in data_new:
    num += i
  return int(num)

### 결과
23524414

## 나의 평균 푸쉬업 개수는?
- 건강을 위해 20일동안 푸쉬업을 했습니다. 20일동안 나의 평균 푸쉬업 개수는 몇개일까요?
    - avg_push(nums)

In [19]:
nums = [6, 2, 5, 7, 9, 6, 3, 8, 11, 15, 9, 16, 16, 12, 18, 20, 25, 27, 23, 31]

In [46]:
# 함수 작성

def avg_push(nums):
  total_sum = 0
  for num in nums:
    total_sum = total_sum + num

  return total_sum/len(nums)
 

In [47]:
print(avg_push(nums))

13.45


### 결과
13.45

## 이게 다 얼마야?
- 제가 No브랜드 과자를 엄청 좋아해요. 장보러 간김에 먹고싶은 과자들을 마구마구 담았는데요. 막상 사고보니 가격과 열량이 궁금하더라구요. 제품 : [가격, 열량] 으로 표시가 되어있는데, 구매 예정 과자의 총 가격, 총 열량은 얼마나 될까요?
    - no_snanks = {'미니비스킷': [2580, 1575], '별사탕건빵': [980, 1025], '바삭한콘칩': [1950, 503], '담백한통감자': [1680, 524], '고르곤졸라소프트콘': [1280, 690]}

In [ ]:
no_snacks = {'미니비스킷': [2580, 1575], '별사탕건빵': [980, 1025], '바삭한콘칩': [1950, 503], '담백한통감자': [1680, 524], '고르곤졸라소프트콘': [1280, 690]}
total_price = 0
total_kcal = 0

In [52]:
# 함수 작성
no_snacks = {'미니비스킷': [2580, 1575], '별사탕건빵': [980, 1025], '바삭한콘칩': [1950, 503], '담백한통감자': [1680, 524], '고르곤졸라소프트콘': [1280, 690]}
total_price = 0
total_kcal = 0
def cal_total_price_kcal(no_snacks, total_price, total_kcal):
  for i in no_snacks:
    total_price += no_snacks[i][0]
    total_kcal += no_snacks[i][1]
  print("총가격 : ", total_price, "원")
  print("총열량 : ", total_kcal, "kcal")
  

In [53]:
cal_total_price_kcal(no_snacks, total_price, total_kcal)

총가격 :  8470 원
총열량 :  4317 kcal


### 결과
총가격 : 8470 원 <br>
총열량 : 4317 kcal